Build Error: macOS Tahoe 26.0.1

Machine: 13-Inch M4 MacBook Air (2025)
OS: Tahoe 26.0.1

After executing 'make setup' the following error occurs, which does not finish the build.

```
Building wheels for collected packages: lru-dict
  Building wheel for lru-dict (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for lru-dict (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [24 lines of output]
      /private/var/folders/k1/rg_1d0m15l31cylc4x7xttkc0000gn/T/pip-build-env-guvs2obj/overlay/lib/python3.9/site-packages/setuptools/dist.py:759: SetuptoolsDeprecationWarning: License classifiers are deprecated.
      !!

              ********************************************************************************
              Please consider removing the following classifiers in favor of a SPDX license expression:

              License :: OSI Approved :: MIT License

              See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
              ********************************************************************************

      !!
        self._finalize_license_expression()
      running bdist_wheel
      running build
      running build_ext
      building 'lru' extension
      creating build/temp.macosx-10.9-universal2-cpython-39
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -iwithsysroot/System/Library/Frameworks/System.framework/PrivateHeaders -iwithsysroot/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/Headers -arch arm64 -arch x86_64 -Werror=implicit-function-declaration -Wno-error=unreachable-code -Wno-error=unused-but-set-variable -Wno-error=cast-function-type-mismatch -Wno-unknown-warning-option -I/Users/eyunghans/Desktop/claude-homeassistant/venv/include -I/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/include/python3.9 -c lru.c -o build/temp.macosx-10.9-universal2-cpython-39/lru.o
      lru.c:629:17: error: incompatible function pointer types initializing 'PyCFunction' (aka 'struct _object *(*)(struct _object *, struct _object *)') with an expression of type 'PyCFunctionWithKeywords' (aka 'struct _object *(*)(struct _object *, struct _object *, struct _object *)') [-Wincompatible-function-pointer-types]
        629 |     {"popitem", (PyCFunctionWithKeywords)LRU_popitem, METH_VARARGS | METH_KEYWORDS,
            |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for lru-dict
Failed to build lru-dict
error: failed-wheel-build-for-install

× Failed to build installable wheels for some pyproject.toml based projects
╰─> lru-dict
make: *** [setup] Error 1
```

--- a/Makefile.dev
+++ b/Makefile.dev
@@ -5,7 +5,7 @@
 include Makefile
 
 # Development-specific variables
-PYTHON = python3.11
+PYTHON = python3.12
 PIP = pip
 VENV_ACTIVATE = source venv/bin/activate
 TEST_DIR = tests

--- a/README-DEV.md
+++ b/README-DEV.md
@@ -96,7 +96,7 @@ make -f Makefile.dev dev-clean-dev
 
 ### Black (Code Formatting)
 - Line length: 88 characters
-- Target Python version: 3.11+
+- Target Python version: 3.12+
 - Automatically formats all Python files
 
 ### isort (Import Sorting)

--- a/README.md
+++ b/README.md
@@ -369,6 +369,16 @@ The entity explorer helps you understand what's available:
 
 ## 🐛 Troubleshooting
 
+### Build Error: lru-dict or Other C Extensions (macOS)
+If you see errors like `Building wheel for lru-dict... error` during `make setup`, you likely have an old Python version from Xcode Command Line Tools.
+
+**Solution**: Install Python 3.12+ via Homebrew:
+```bash
+brew install python@3.12
+export PATH="/opt/homebrew/bin:$PATH"
+```
+Then run `make setup` again. Home Assistant 2024.x requires Python 3.12+.
+
 ### Validation Errors
 1. Check YAML syntax first: `. venv/bin/activate && python tools/yaml_validator.py`
 2. Verify entity references: `. venv/bin/activate && python tools/reference_validator.py`

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -9,7 +9,7 @@ description = "Home Assistant configuration management and validation tools"
 authors = [
     {name = "Home Assistant Config Management"}
 ]
-requires-python = ">=3.11"
+requires-python = ">=3.12"
 dependencies = [
     "homeassistant>=2024.1.0",
     "voluptuous>=0.14.0",
@@ -32,7 +32,7 @@ dev = [
 
 [tool.black]
 line-length = 88
-target-version = ['py311']
+target-version = ['py312']
 include = '\.pyi?$'
 extend-exclude = '''
 /(
@@ -68,7 +68,7 @@ extend_skip_glob = ["venv/*", "config/*", "temp/*"]
 # Note: flake8 configuration is in .flake8 file (doesn't support pyproject.toml)
 
 [tool.mypy]
-python_version = "3.11"
+python_version = "3.12"
 warn_unused_configs = true
 # Relaxed settings for existing codebase
 disallow_untyped_defs = false

--- a/setup-mac.sh
+++ b/setup-mac.sh
@@ -22,21 +22,34 @@ command_exists() {
 
 echo "🔍 Checking prerequisites..."
 
-# Check if Python 3.8+ is available
+# Check if Python 3.12+ is available (required by Home Assistant 2024.x)
 if ! command_exists python3; then
     echo "❌ Python 3 is not installed."
-    echo "Please install Python from https://www.python.org/downloads/"
-    echo "Or use Homebrew: brew install python3"
+    echo ""
+    echo "Install Python 3.12+ via Homebrew (recommended):"
+    echo "   brew install python@3.12"
+    echo ""
+    echo "Then run this script again."
     exit 1
 fi
 
-# Check Python version
+# Check Python version - Home Assistant 2024.x requires Python 3.12+
 PYTHON_VERSION=$(python3 -c 'import sys; print(".".join(map(str, sys.version_info[:2])))')
-REQUIRED_VERSION="3.8"
 
-if ! python3 -c "import sys; exit(0 if sys.version_info >= (3, 8) else 1)"; then
-    echo "❌ Python $PYTHON_VERSION found, but Python 3.8+ is required."
-    echo "Please upgrade Python from https://www.python.org/downloads/"
+if ! python3 -c "import sys; exit(0 if sys.version_info >= (3, 12) else 1)"; then
+    echo "❌ Python $PYTHON_VERSION found, but Python 3.12+ is required."
+    echo ""
+    echo "Home Assistant 2024.x requires Python 3.12 or newer."
+    echo "The system Python from Xcode Command Line Tools is typically too old"
+    echo "and may cause build errors with C extensions."
+    echo ""
+    echo "Install Python 3.12+ via Homebrew:"
+    echo "   brew install python@3.12"
+    echo ""
+    echo "After installation, ensure Homebrew Python is in your PATH:"
+    echo "   export PATH=\"/opt/homebrew/bin:\$PATH\""
+    echo ""
+    echo "Then run this script again."
     exit 1
 fi
 

